openedx/features/personalized_learner_schedules/call_to_action.py

```diff
@@ -81,6 +81,17 @@ def _is_block_shiftable(xblock):
             PersonalizedLearnerScheduleCallToAction._log_past_due_warning(type(xblock).__name__)
             is_past_due = xblock.is_past_due
 
+        # This is used to replicate the logic used in common/lib/xmodule/xmodule/vertical_block.py
+        # We only want to show the banner if there are any problems that are graded and scored.
+        # Note: we will still show the CTA at the xblock level (next to the submit button) regardless
+        # of graded and scored status
+        if category == cls.VERTICAL_BANNER:
+            graded = getattr(xblock, 'graded', False)
+            has_score = getattr(xblock, 'has_score', False)
+            weight = getattr(xblock, 'weight', 1)
+            scored = has_score and (weight is None or weight > 0)
```

Comment:
I'm also tired of making this same calculation over and over. But not this PR's fight.